MatrixSDK/Crypto/Algorithms/Megolm/MXMegolmEncryption.m

```diff
@@ -464,8 +490,7 @@ - (MXHTTPOperation*)reshareKey:(NSString*)sessionId
     }
     
     // Get the chain index of the key we previously sent this device
-    MXOutboundSessionInfo *obSessionInfo = outboundSessions[sessionId];
-    NSNumber *chainIndex = [obSessionInfo.sharedWithDevices objectForDevice:deviceId forUser:userId];
+    NSNumber *chainIndex = [crypto.store messageIndexForSharedDeviceInRoomWithId:roomId sessionId:sessionId userId:userId deviceId:deviceId];
```

Comment:
`self.outboundSessions` look useless now. Do you think we can remove it?

Reply:
`self.outboundSessions` is useful especially to establish the share map when the user sends a message and for its helper methods.

Reply:
It was useful for caching used megolm sessions in memory. With this PR, we only write to it and never read from it.

Reply:
Removed. Sorry for the misunderstanding.